star-rx72n-firmware/lib/rx_hal/src/adc.c

```
@@ -106,6 +113,38 @@ static volatile rx_s12ad_regs_t* internal_get_adc_base(uint8_t unit)
   }
 }
 
+static rx_err_t
+internal_configure_adc_unit(uint8_t unit, volatile rx_s12ad_regs_t* adc, uint8_t bits)
+{
+  uint16_t adcer;
+
+  system_regs()->prcr = k_rx_prcr_unlock_prc1_prc3;
+
+  if (unit == k_adc_unit_0) {
```

Clang (14.0.6)

[warning] if with identical then and else branches

(bugprone-branch-clone)